### PR TITLE
chore: [#187815961] fix failing profile tests

### PR DESCRIPTION
**Pull request by FarazA22**
### Ticket

https://www.pivotaltracker.com/story/show/187815961

### Approach
There was an issue with how profile tests were loading municipalities leading to a mui autocomplete component error. We updated the test logic to load all possible municipalities from userData. 

Separately, we removed the logic previously added that prevented the failing of tests when the address in oscar profile was added.

## Code author checklist

- [x] I have rebased this branch from the latest main branch
- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see [CMS Additions in Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po/edit#heading=h.fu5jdsrcqxbh))
- [x] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: web/test/pages/profile/profile-owning.test.tsx [hunk 1]
```diff
@@ -139,25 +133,23 @@ describe("profile - owning existing business", () => {
         businessPersona: "OWNING",
         industryId: "generic",
         legalStructureId: "limited-liability-company",
-        municipality: undefined,
       }),
       onboardingFormProgress: "COMPLETED",
-      formationData: formationWithUndefinedMunicipality,
     });
-    const newark = generateMunicipality({ displayName: "Newark" });
+    const randomMuni = generateMunicipality({ displayName: "Newark" });
```
Comment (adidner) [inline web/test/pages/profile/profile-owning.test.tsx:139]: pebble - I don't love the abbreviation here. Would personally prefer randomMunicipality

File: web/test/pages/profile/profile-owning.test.tsx [hunk 2]
```diff
@@ -211,6 +203,7 @@ describe("profile - owning existing business", () => {
   });
 
   it("prefills form from existing user data", () => {
+    const randomMuni = generateMunicipality({});
```
Comment (adidner) [inline web/test/pages/profile/profile-owning.test.tsx:206]: same as above

File: web/test/pages/profile/profile-owning.test.tsx [hunk 3]
```diff
@@ -330,9 +319,9 @@ describe("profile - owning existing business", () => {
   });
 
   it("returns user to dashboard from un-saved changes modal", async () => {
-    const newark = generateMunicipality({ displayName: "Newark" });
-    renderPage({ business, municipalities: [newark] });
-    selectByText("Location", newark.displayName);
+    const randomMuni = generateMunicipality({});
```
Comment (adidner) [inline web/test/pages/profile/profile-owning.test.tsx:322]: also same

File: web/test/pages/profile/profile-owning.test.tsx [hunk 4]
```diff
@@ -622,6 +603,7 @@ describe("profile - owning existing business", () => {
       });
 
       it("displays alert when address city is selected then blurred", async () => {
+        const randomMuni = generateMunicipality({});
```
Comment (adidner) [inline web/test/pages/profile/profile-owning.test.tsx:606]: same

File: web/test/pages/profile/profile-shared.test.tsx [hunk 5]
```diff
@@ -274,38 +263,36 @@ describe("profile - shared", () => {
...
+      return expect(currentBusiness().profileData.municipality).toEqual(randomMuni);
     });
     expect(
       mockAnalytics.event.profile_location_question.submit.location_entered_for_first_time
     ).toHaveBeenCalled();
   });
 
   it("does not send analytics when municipality already existed", async () => {
-    const newark = generateMunicipality({ displayName: "Newark" });
+    const randomMuni = generateMunicipality({});
```
Comment (adidner) [inline web/test/pages/profile/profile-shared.test.tsx:283]: same

File: web/test/pages/profile/profile-starting.test.tsx [hunk 6]
```diff
@@ -718,6 +709,7 @@ describe("profile - starting business", () => {
   });
 
   it("prefills form from existing user data", () => {
+    const randomMuni = generateMunicipality({});
```
Comment (adidner) [inline web/test/pages/profile/profile-starting.test.tsx:712]: same

File: web/test/pages/profile/profile-starting.test.tsx [hunk 7]
```diff
@@ -380,29 +371,29 @@ describe("profile - starting business", () => {
...
           operatingPhase: params.operatingPhase,
           businessPersona: "STARTING",
         }),
       });
-      renderPage({ municipalities: [newark], business });
+      renderPage({ business });
     };
 
     it("locks the location field when it is populated and tax filing state is SUCCESS", () => {
+      const randomMuni = generateMunicipality({});
```
Comment (adidner) [inline web/test/pages/profile/profile-starting.test.tsx:385]: same

File: web/test/pages/profile/profile-starting.test.tsx [hunk 8]
```diff
@@ -292,10 +283,10 @@ describe("profile - starting business", () => {
       },
     };
     const inputFieldName = getBusinessProfileInputFieldName(initialBusiness);
-    const newark = generateMunicipality({ displayName: "Newark" });
-    renderPage({ business: initialBusiness, municipalities: [newark] });
+    const randomMuni = generateMunicipality({ displayName: "Newark" });
```
Comment (adidner) [inline web/test/pages/profile/profile-starting.test.tsx:286]: same

**Review by adidner — APPROVED** [review]
Approved. Looked good, I would prefer a minor naming change.

File: web/test/pages/profile/profile-owning.test.tsx [hunk 9]
```diff
@@ -403,19 +386,19 @@ describe("profile - owning existing business", () => {
 
   describe("Location Section", () => {
     it("locks the location field if it is populated and tax filing state is SUCCESS", () => {
+      const randomMuni = generateMunicipality({});
```
Comment (seidior) [inline web/test/pages/profile/profile-owning.test.tsx:389]: [dust] same as Aaron's comments

File: web/test/pages/profile/profile-owning.test.tsx (hunk 1 above)
Comment (FarazA22) [reply web/test/pages/profile/profile-owning.test.tsx]: I made the change across the board

File: web/test/pages/profile/profile-owning.test.tsx (hunk 9 above)
Comment (FarazA22) [reply web/test/pages/profile/profile-owning.test.tsx]: I made the change across the board

**Pull request merged**